feat(ROADMAP): updates to public roadmap 🚀 (#28302)
aws/aws-cdk: ROADMAP.md

The AWS CDK team values feedback and guidance from its community of users, although final decisions on inclusion into the project will be made by AWS. We determine the high-level direction for our open roadmap based on customer feedback and popularity (👍🏽 and comments), security and operational impacts, and business value. Where features don’t meet our goals and longer-term strategy, we will communicate that clearly and openly as quickly as possible with an explanation of why the decision was made.

## FAQs

**Q: Why did you build an open roadmap?**
A: Your feedback and suggestions would help in ensuring that we are working on the most important and impactful issues. And if you are making decisions and plans based on what we are developing, this will provide insights on what is coming down the road for the CDK.

**Q: Why are there no dates on your roadmap?**
A: Because security and operational stability are our highest priorities, the above new features cannot be provided specific target dates. The roadmap is subject to change at any time, and roadmap issues in this repository do not guarantee a feature will be launched as proposed.

**Q: Is everything on the roadmap?**

> A: We will publish high-level direction that is within the scope of the CDK. Minor features and performance improvement tasks are not on the roadmap currently, but we are constantly trying to improve the roadmap so please leave your suggestions [here](https://github.com/aws/aws-sam-cli/issues/3267).

Reviewer: This appears to be a link to a closed issue on the aws-sam-cli repository, not the CDK. Is that the right place to leave comments and suggestions regarding AWS CDK?

## Disclaimer

The AWS CDK team values feedback and guidance from its community of users, although final decisions on inclusion into the project will be made by AWS. We determine the high-level direction for our open roadmap based on customer feedback and popularity (👍🏽 and comments), security and operational impacts, and business value. Where features don’t meet our goals and longer-term strategy, we will communicate that clearly and openly as quickly as possible with an explanation of why the decision was made.

## FAQs

**Q: Why did you build an open roadmap?**
A: Your feedback and suggestions would help in ensuring that we are working on the most important and impactful issues. And if you are making decisions and plans based on what we are developing, this will provide insights on what is coming down the road for the CDK.

**Q: Why are there no dates on your roadmap?**
A: Because security and operational stability are our highest priorities, the above new features cannot be provided specific target dates. The roadmap is subject to change at any time, and roadmap issues in this repository do not guarantee a feature will be launched as proposed.

> 
> 

Reviewer 2: ```suggestion
```

**Q: Is everything on the roadmap?**
A: We will publish high-level direction that is within the scope of the CDK. Minor features and performance improvement tasks are not on the roadmap currently, but we are constantly trying to improve the roadmap so please leave your suggestions [here](https://github.com/aws/aws-sam-cli/issues/3267).

**Q: How can I provide feedback or ask for more information?**
A: When in doubt, please create an issue! Issues will be reviewed and/or forwarded appropriately. A great time to provide feedback is when the project is in Exploration, RFC stage, or when the feature is in beta release. As always, we listen to your feedback and adapt our plans if needed.

**Q: Can I 👍🏽 existing issues?**
A: We strongly encourage you to do so, as it helps us understand which issues will have the widest impact. You can navigate to the issue details page and add a reaction (👍🏽).

> Q: **How can I request a feature be added to the roadmap?**

Author: ```suggestion
**Q: How can I request a feature be added to the roadmap?**
```

# AWS CDK Roadmap

> The roadmap priorities for the AWS CDK (Cloud Development Kit) are informed by what we hear from customers and interactions on Github, CDK.dev Slack, Stack Overflow, and Twitter. This document outlines the high level direction we are working towards, and for each project there is a tracking issue where you can leave feedback. We update this doc on a quarterly basis to reflect any changing priorities. 

Reviewer 2: I think we're missing a sentence introducing what the AWS CDK Roadmap is

Reviewer 2: cdk.dev instead of CDK.dev?

Reviewer 2: also nit: we should split up paragraphs into individual lines so that they are easier to review and read in raw text.

Follow <a href="https://www.youtube.com/@CDK-Live">CDK Live!</a> and <a href="https://cdk.dev/">cdk.dev</a> to learn what's new in AWS CDK.

> Security and stability of the CDK is a top priority. If you think you’ve found a potential security issue, please do not post it in the issues. Instead, please follow the instructions <a href="https://aws.amazon.com/security/vulnerability-reporting/">here</a> or directly email <a href="mailto:aws-security@amazon.com">AWS security</a>.

Reviewer 3: ```suggestion
Security and stability of the CDK is a top priority. If you think you’ve found a potential security issue, please do not post it as an issue or a discussion thread in this repository. Instead, please follow the instructions <a href="https://aws.amazon.com/security/vulnerability-reporting/">here</a> or directly email <a href="mailto:aws-security@amazon.com">AWS security</a>.
```

[aws-cdk]: https://github.com/aws/aws-cdk
[aws-cdk-rfcs]: https://github.com/aws/aws-cdk-rfcs
[jsii]: https://github.com/aws/jsii

## Our Core Tenets (which guide prioritization decisions)

* **We empower CDK builders to innovate with confidence, without sacrificing security.** The CDK provides customers with conviction by providing a clear and streamlined direction to building secure, modernized, industry leading innovations with AWS.
* **Productivity through simplicity.** The CDK measurably improves developer productivity in building with AWS. We make Infrastructure as Code simple to define, understand, change, and troubleshoot.
* **Be transparent.** The AWS CDK team’s current work should be easily visible.
* **Listen to customers.** Allow them to participate in design decisions and to vote on and propose new AWS CDK features. We will periodically reprioritize the roadmap based on customer feedback.

## List of Annotations

>   <table>

Reviewer 2: Same for this -- there is a canonical way to do this in markdown syntax, which I prefer because it is easier to maintain and update the table in the future.

https://www.markdownguide.org/extended-syntax/#tables

* 👂🏽 <a href="https://github.com/aws/aws-cdk-rfcs/issues/228">CDK CLI Triggers</a> - CLI enhancements are also being considered for post command hooks. 

### L2 Abstractions
* 🔍 <a href="https://github.com/aws/aws-cdk-rfcs/issues/491">CloudFront Origin Access Control L2</a>

We are currently investigating other L2s to build out next. Feel free to create an RFC to request.

## Community Engagement
We would love to hear from you on how the CDK operates today and how it should grow in the future. To report a bug or create a small feature request, please <a href="https://github.com/aws/aws-cdk/issues/new/choose">create an issue here</a>. If you are seeking to request a change in strategic direction or make a CDK core framework change, please <a href="https://github.com/aws/aws-cdk-rfcs/issues/new/choose">create a Request for Comments (RFC) ticket here</a>. If you are ever unsure about where your feature request should live, it is best to follow the first link within the aws-cdk repo.

> <img width="1174" alt="image" src="https://github.com/aws/aws-cdk/assets/142322013/ea006330-caa7-4c00-8eba-8e8fe379ef6b">

Reviewer 2: This can also be done in markdown syntax, look at the raw markdown to see what i did: ![image](https://github.com/aws/aws-cdk/assets/142322013/ea006330-caa7-4c00-8eba-8e8fe379ef6b)

* [Videos] <a href="https://www.youtube.com/@CDK-Live">CDK Live!</a>
* [CDK Day] <a href="https://www.youtube.com/watch?v=qlUR5jVBC6c">Track 1 (English) for CDK Day 2023</a>
* [CDK Day] <a href="https://www.youtube.com/watch?v=b-nSH18gFQk">Track 2 (English) for CDK Day 2023</a>
* [CDK Day] <a href="https://www.youtube.com/watch?v=ZAQC-cOXL4M">Track 3 (Spanish) for CDK Day 2023</a>
* [re:Invent Content] Search through <a href="https://www.youtube.com/@amazonwebservices">all content here</a>

## 2024 Updates

Stay tuned for more updates for 2024!

## Disclaimer

The AWS CDK team values feedback and guidance from its community of users, although final decisions on inclusion into the project will be made by AWS. We determine the high-level direction for our open roadmap based on customer feedback and popularity (👍🏽 and comments), security and operational impacts, and business value. Where features don’t meet our goals and longer-term strategy, we will communicate that clearly and openly as quickly as possible with an explanation of why the decision was made.

## FAQs

> Q: Why did you build an open roadmap?

Reviewer 2: The QandA section doesn't render the way you're expecting, I think. You need a space between the lines for a space to show up in the markdown render.
